cmd/create/externalauthproviders/cmd.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalauthproviders
+
+import (
+	"os"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/externalauthproviders"
+	"github.com/openshift/rosa/pkg/input"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var externalAuthProvidersArgs *externalauthproviders.ExternalAuthProvidersArgs
+
+const argsPrefix string = ""
+
+var Cmd = &cobra.Command{
+	Use:     "external-auth-provider",
+	Aliases: []string{"externalauthproviders", "externalauthprovider", "external-auth-provider"},
+	Short:   "Create an external authentication provider for a cluster.",
+	Long:    "Configuring the cluster with external authentication configuration instead of internal oidc provider. ",
+	Example: `  # Interactively create an external authentication provider to a cluster named "mycluster"
+  rosa create external-auth-provider --cluster=mycluster --interactive`,
+	Run:    run,
+	Hidden: true,
+	Args:   cobra.NoArgs,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	ocm.AddClusterFlag(Cmd)
+	interactive.AddFlag(flags)
+	externalAuthProvidersArgs = externalauthproviders.AddExternalAuthProvidersFlags(Cmd, argsPrefix)
+}
+
+func run(cmd *cobra.Command, _ []string) {
```

Comment:
@chenz4027 Could you follow the same pattern as the other commands where you delegate to a `runWithRuntime` function. With the `os.Exit(1)` calls, there is no way for us to test this function.